Client/Frontend/Browser/Toolbars/BottomToolbar/Menu/Bookmarks/BookmarkDetailsView.swift

```diff
@@ -88,6 +94,45 @@ class BookmarkDetailsView: AddEditHeaderView, BookmarkFormFieldsProtocol {
     // MARK: - Delegate actions
     
     @objc func textFieldDidChange(_ textField: UITextField) {
-        delegate?.correctValues(validationPassed: validateFields())
+        delegate?.correctValues(validationPassed: validateFields() || validateCodeFields())
+    }
+    
+    private func validateTitle(_ title: String?) -> Bool {
+        guard let title = title else { return false }
+        return !title.isEmpty
+    }
+    
+    private func validateCodeFields() -> Bool {
+        let uriScheme = "javascript:"
+        
+        let title = titleTextField.text
+        guard let urlTextField = urlTextField else { return validateTitle(title) }
+        guard var javascriptCode = urlTextField.text else { return false }
+        
+        // All bookmarklets must begin with `javascript` URI scheme..
+        if !javascriptCode.hasPrefix(uriScheme) {
+            return false
+        }
+        
+        javascriptCode = String(javascriptCode.dropFirst(uriScheme.count))
+        
+        if javascriptCode.isEmpty {
+            return false
+        }
+        
+        // A bookmarklet is considered valid if it's code is valid JS.
+        // Bookmarklets MIGHT invoke some security flaws allowing the user to run arbitrary
+        // JS in the browser though.
```

Comment:
maybe specify in the comment that this is running in the webpage context (as opposed to browser context)

assuming this is implemented correctly, the security implications of bookmarklets should only be that they introduce a new way for self-XSS